packages/plugin-ext/src/main/node/plugin-deployer-impl.ts

```diff
@@ -298,13 +298,12 @@ export class PluginDeployerImpl implements PluginDeployer {
         const pluginPaths = [...acceptedBackendPlugins, ...acceptedHeadlessPlugins].map(pluginEntry => pluginEntry.path());
         this.logger.debug('local path to deploy on remote instance', pluginPaths);
 
-        const deployments = await Promise.all([
-            // headless plugins are deployed like backend plugins
-            this.pluginDeployerHandler.deployBackendPlugins(acceptedHeadlessPlugins),
-            // start the backend plugins
-            this.pluginDeployerHandler.deployBackendPlugins(acceptedBackendPlugins),
-            this.pluginDeployerHandler.deployFrontendPlugins(acceptedFrontendPlugins)
-        ]);
+        const deployments = [];
+        // start the backend plugins
+        deployments.push(await this.pluginDeployerHandler.deployBackendPlugins(acceptedBackendPlugins));
+        // headless plugins are deployed like backend plugins
+        deployments.push(await this.pluginDeployerHandler.deployBackendPlugins(acceptedHeadlessPlugins));
+        deployments.push(await this.pluginDeployerHandler.deployFrontendPlugins(acceptedFrontendPlugins));
```

Comment:
Minor: Since we only count the number of deployments anyway, wouldn't something like this be more reasonable? And then simply return with `deployments.length`.

```suggestion
        const deployments = [];
        // start the backend plugins
        deployments.push(...await this.pluginDeployerHandler.deployBackendPlugins(acceptedBackendPlugins));
        // headless plugins are deployed like backend plugins
        deployments.push(...await this.pluginDeployerHandler.deployBackendPlugins(acceptedHeadlessPlugins));
        deployments.push(...await this.pluginDeployerHandler.deployFrontendPlugins(acceptedFrontendPlugins));
```

Reply:
The `deploy<X>Plugins()` calls return a count or undefined, not the actual plugins and I really don't want to invest time in what is basically a workaround for a larger problem.

Reply:
Alright, I see. Ok